README.md

```diff
@@ -1458,6 +1458,91 @@ map.addEntryListener(mapEventListener).then(function () {
+
+An entry processor is a good option if you perform bulk processing on an `IMap`. Usually you perform a loop of keys-- executing `IMap.get(key)`, mutating the value, and finally putting the entry back in the map using `IMap.put(key,value)`. If you perform this process from a client or from a member where the keys do not exist, you effectively perform two network hops for each update: the first to retrieve the data and the second to update the mutated value.
+
+If you are doing the process described above, you should consider using entry processors. An entry processor executes a read and updates upon the member where the data resides. This eliminates the costly network hops described above.
+
+> Note: Entry processor is meant to process a single entry per call. Processing multiple entries and data structures in an entry processor is not supported as it may result in deadlocks in the server side.
+
+Hazelcast sends the entry processor to each cluster member and these members apply it to map entries. Therefore, if you add more members, your processing completes faster.
+
+#### Processing Entries
+
+The functions below are in the `IMap` functions for entry processing.
+
+- `executeOnKey` processes an entry mapped by a key.
+
+- `executeOnKeys` processes entries mapped by a list of keys.
+
+- `executeOnEntries` can process all entries in a map with a defined predicate. Predicate is optional.
+
+In the Node.js client, an `EntryProcessor` should be `IdentifiedDataSerializable` or `Portable` because server should be able to deserialize it to process.
+
+An `EntryProcessor` which is `IdentifiedDataSerializable` example is shown below.
+
+```javascript
+function IdentifiedEntryProcessor(value) {
+    this.value = value;
+}
+
+IdentifiedEntryProcessor.prototype.readData = function (inp) {
+    this.value = inp.readUTF();
+};
+
+IdentifiedEntryProcessor.prototype.writeData = function (outp) {
+    outp.writeUTF(this.value);
+};
+
+IdentifiedEntryProcessor.prototype.getFactoryId = function () {
+    return 5;
+};
+
+IdentifiedEntryProcessor.prototype.getClassId = function () {
+    return 1;
+};
+```
+
+You should also create a factory for the processor like below.
+
+```javascript
+function EntryProcessorFactory() {
+
+}
+
+EntryProcessorFactory.prototype.create = function (type) {
+    if (type === 1) {
+        return new IdentifiedEntryProcessor();
+    }
+    return null;
+};
+```
+
+Then, You need to implement the Java equivalent of your entry processor and its factory and create your own compiled class or JAR files. For adding your own compiled class or JAR files, please see [Adding User Library to CLASSPATH section](#adding-user-library-to-classpath). Note that you need to configure the `hazelcast.xml` to add your factory. For more details about the Java implementation of the `EntryProcessor`, see [Entry Processor section](https://docs.hazelcast.org/docs/latest/manual/html-single/index.html#entry-processor) in the IMDG Reference Manual.
+
```

Comment:
I think the following is also a crucial information for the users:
`The code that runs on entries is implemented in Java on the server side. Client side entry processor is used to specify which entry processor should be called and to configure the entry processor.`

My suggestion is to add this here. Also we need the actual Java implementation of the processor and hazelcast.xml in the sample.